src/Connection.php

```diff
@@ -371,6 +379,21 @@ public function connect()
         );
 
         if ($this->_conn !== null) {
+            if (! class_exists(ConnectionLossAwareHandler::class) || null === self::$checkFrequency) {
+                return false;
+            }
+
+            if (! self::$isChecking) {
+                if (null === self::$lastCheckedAt || time() - self::$lastCheckedAt >= self::$checkFrequency) {
+                    self::$isChecking = true;
+
+                    ConnectionLossAwareHandler::reconnectOnFailure($this);
```

Comment:
Maybe I could copy the code directly here instead of having a dependency to Symfony